aws_lambda_powertools/utilities/batch/base.py

```diff
@@ -87,19 +88,6 @@ def _clean(self):
         """
         raise NotImplementedError()
 
-    @abstractmethod
-    def _process_record(self, record: dict):
-        """
-        Process record with handler.
-        """
-        raise NotImplementedError()
-
-    def process(self) -> List[Tuple]:
-        """
-        Call instance's handler for each record.
-        """
-        return [self._process_record(record) for record in self.records]
```

Comment:
This will break custom processors created by customers, since `batch_processor` decorator will call `processor.process()` -- and TIL we didn't have tests for a custom processor, otherwise it'd have signaled it.

https://awslabs.github.io/aws-lambda-powertools-python/2.3.1/utilities/batch/#create-your-own-partial-processor